### PR TITLE
Feat: change route in place from id to position

### LINKED ISSUE #174: Change route in place from id to position

### PR DESCRIPTION
**Pull request by braianj**

### CODE REVIEW

File: src/entities/Social/routes.ts [hunk 1]
```diff
@@ -27,12 +28,23 @@ async function readFile(req: Request) {
...
 export async function injectPlaceMetadata(req: Request) {
-  const id = String(req.query.id || "")
+  const position = String(req.query.position || "")
   const page = await readFile(req)
-  if (isUUID(id)) {
-    const place = await PlaceModel.findByIdWithAggregates(id, {
-      user: undefined,
-    })
+  const parsedPosition = validPosition(position)
+  console.log(parsedPosition)
```
Comment (2fd) [inline src/entities/Social/routes.ts:34]: 🤡 where's `console.log`

File: src/entities/Social/routes.ts [hunk 2]
```diff
@@ -27,12 +28,23 @@ async function readFile(req: Request) {
 }
 
 export async function injectPlaceMetadata(req: Request) {
-  const id = String(req.query.id || "")
+  const position = String(req.query.position || "")
```
Comment (2fd) [inline src/entities/Social/routes.ts:31]: We should keep supporting `id` and redirect all users to the `position` version

File: src/utils/position/validPosition.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import { isInsideWorldLimits } from "@dcl/schemas/dist/dapps/world"
+
+function validPosition(positionString?: string | null) {
```
Comment (2fd) [inline src/utils/position/validPosition.ts:3]: we should iterate this name, isn't completely clear

**Review by 2fd — CHANGES_REQUESTED** [review]

File: src/pages/place.tsx [hunk 4]
```diff
@@ -30,14 +31,34 @@ export default function PlacePage() {
...
+
+  const placeMemo = useMemo(
+    () => (!placeRetrived ? [[]] : [[placeRetrived]]),
+    [placeRetrived]
+  )
+
+  useEffect(() => {
+    if (
+      placeRetrived &&
+      placeRetrived.base_position &&
```
Comment (2fd) [inline src/pages/place.tsx:52]: Is it possible for `base_position` to be empty if the places exist? If it is, we should look for a more robust alternative to represent the place URL

File: src/pages/place.tsx [hunk 5]
```diff
@@ -61,26 +82,25 @@ export default function PlacePage() {
...
-
   const [ff] = useFeatureFlagContext()
 
   if (ff.flags[FeatureFlags.Maintenance]) {
     return <MaintenancePage />
   }
 
-  if (!loading && !place) {
+  if (
+    placeRetrivedState.version > 0 &&
```
Comment (2fd) [inline src/pages/place.tsx:100]: we should consider adding a new prop like `firstLoading` in `decentraland-gatsby`

**Review by 2fd — CHANGES_REQUESTED** [review]

File: src/pages/place.tsx (hunk 5 above)
Comment (braianj) [reply src/pages/place.tsx:100]: 👍

File: src/pages/place.tsx [hunk 6]
```diff
@@ -30,14 +32,34 @@ export default function PlacePage() {
...
+  )
+
+  useEffect(() => {
+    if (
+      placeRetrived &&
+      toCanonicalPosition(placeRetrived.base_position) !==
+        params.get("position")
+    ) {
+      navigate(
+        locations.place(toCanonicalPosition(placeRetrived.base_position)!),
```
Comment (2fd) [inline src/pages/place.tsx:57]:
```suggestion
        locations.place(placeRetrived.base_position!),
```

File: src/modules/locations.ts [hunk 7]
```diff
@@ -68,7 +68,8 @@ export function fromPlacesOptions(
 
 export default {
   home: () => API.url(GATSBY_BASE_URL, "/"),
-  place: (id: string) => API.url(GATSBY_BASE_URL, "/place/", { id }),
+  place: (position: string) =>
+    API.url(GATSBY_BASE_URL, "/place/", { position }),
```
Comment (2fd) [inline src/modules/locations.ts:72]:
```suggestion
    API.url(GATSBY_BASE_URL, "/place/", { position: toCanonicalPosition(position) }),
```

**Review by 2fd — APPROVED** [review]

**Pull request merged**